src/wallet/bdk.rs

```diff
@@ -308,77 +366,205 @@ impl Wallet {
                 let locktime = bond.lock_time;
                 println!(
                     "[{}] locktime={}",
-                    Address::from_script(&bond.script_pub_key(), self.store.network).unwrap(),
+                    Address::from_script(&bond.script_pub_key(), self.network).unwrap(),
                     locktime
                 );
             }
         }
         Ok(())
     }
 
-    pub fn init(
+    pub fn init<E: IntoWalletDescriptor + Hash + Debug>(
```

Comment:
```suggestion
    pub fn init<E>(
    path: impl AsRef<Path>,
    seed_phrase: &str,
    passphrase: &str,
    network: Network,
    descriptors: WalletDescriptors<E>,
    rpc_config: &RPCConfig,
    wallet_birthday: Option<u64>,
) -> Result<Self, NewError>
where
    E: IntoWalletDescriptor + Hash + Debug,

```
The method signature can be changed to this. 
Can have a wallet descriptor struct like ths
```

pub struct WalletDescriptors<E>
where
    E: IntoWalletDescriptor + Hash + Debug,
{
    pub external_descriptor: E,
    pub internal_descriptor: E,
    pub fidelity_descriptor: E
    pub swapcoin_descriptor: E,
    pub contract_descriptor: E,
}

```